clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -300,9 +301,13 @@ void runOnce() {
             try {
                 transactionManager.maybeResolveSequences();
 
+                RuntimeException lastError = transactionManager.lastError();
+                if (transactionManager.hasAbortableError() && shouldHandleAuthorizationError(lastError)) {
+                    return;
```

Comment:
I'm thinking not, because we aren't adding a new producer.  @jolshan thoughts?

Reply:
Sorry -- I think if we have an error we'd still want to bump the epoch right?

Reply:
(Assuming this is an indempotent producer and not a transactional one)

Reply:
Right this is an idempotent producer, thanks for catching the epoch bump, I was wondering about that too.

Reply:
Did we determine that we will end up bumping the epoch? It doesn't look like it is done here so is it somewhere else?

Reply:
FWIW: I think in the context of this ticket, we are trying to avoid poisoning the client when the client is unable to re-authenticate upon startup. The fix wants to continue to retry the request until the permission is fixed. So the handler should handle the epoch bump.

Reply:
Thanks Philip -- I think I forgot that this was the initProducerId call -- so we don't really have an epoch yet. 😅 We set to 0 after getting the producer ID.